pinot-query-planner/src/main/java/org/apache/pinot/query/parser/CalciteRexExpressionParser.java

```diff
@@ -199,6 +203,12 @@ private static Expression compileFunctionExpression(RexExpression.FunctionCall r
         return compileOrExpression(rexCall, pinotQuery);
       case OTHER_FUNCTION:
         functionName = rexCall.getFunctionName();
+        // Special handle for leaf stage multi-value columns, as the default behavior for filter and group by is not
+        // sql standard, so need to use `array_to_mv` to convert the array to v1 multi-value column for behavior
+        // consistency meanwhile not violating the sql standard.
+        if (ARRAY_TO_MV_ALIAS.contains(functionName)) {
```

Comment:
Is function name always upper case?

Reply:
Let's canonicalized and just do `ARRAY_TO_MV`